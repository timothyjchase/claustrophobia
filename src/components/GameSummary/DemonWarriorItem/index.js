import React from 'react'
import PropTypes from 'prop-types'
import renderHTML from 'react-render-html'
import { Button, Label, List, Popup, Rating } from 'semantic-ui-react'

const DemonWarriorItem = ({ warrior }) => {
  if (warrior.numberInPlay)
    return (
      <List.Item>
        <List.Content>
          <Button
            icon="remove"
            floated="right"
            size="tiny"
            onClick={warrior.onRemove}
            compact
            basic
          />
          <List.Header as="a">
            <Label circular horizontal>
              {warrior.numberInPlay}
            </Label>
            <Popup
              inverted
              trigger={(
                <span>
                  {`${warrior.name}${warrior.numberInPlay > 1 ? 's' : ''}`}
                </span>
)}
            >
              <div>
                <Label content={`${warrior.movement} MVT`} />
                {' '}
                <Label content={`${warrior.combat} CBT`} />
                {' '}
                <Label content={`${warrior.defense} DEF`} />
                {!!warrior.rules && <div>{renderHTML(warrior.rules)}</div>}
              </div>
            </Popup>
            {warrior.health > 1 && (
              <div style={{ paddingLeft: '30px' }}>
                <Rating
                  icon="heart"
                  defaultRating={0}
                  maxRating={warrior.health}
                  size="tiny"
                />
              </div>
            )}
          </List.Header>
        </List.Content>
      </List.Item>
    )
  return null
}

DemonWarriorItem.propTypes = {
  warrior: PropTypes.object.isRequired,
}

export default DemonWarriorItem
