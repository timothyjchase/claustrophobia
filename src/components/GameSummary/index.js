import React from 'react'
import PropTypes from 'prop-types'
import { Grid, List, Segment } from 'semantic-ui-react'
import DieImage from '../DieImage'
import DemonWarriorItem from './DemonWarriorItem'

const GameSummary = ({ turn, demonDice, threatDice, warriors }) => {
  const demonWarriorCount = warriors.reduce(
    (result, warrior) => result + warrior.numberInPlay,
    0,
  )
  return (
    <Segment style={{ padding: '10px 10px 10px 10px' }}>
      <Grid stackable={false}>
        <Grid.Row stretched>
          <Grid.Column width={5} verticalAlign="middle">
            <div>
              <strong>
                Turn:
                {turn}
              </strong>
              <br />
              <DieImage
                style={{
                  display: 'inline',
                  width: '30px',
                  height: '30px',
                  margin: '5px 4px 0px 0px',
                }}
                type="DEMON"
                value={demonDice}
              />
              <DieImage
                style={{
                  display: 'inline',
                  width: '30px',
                  height: '30px',
                  margin: '5px 0px 0px 4px',
                }}
                type="THREAT"
                value={threatDice}
              />
            </div>
          </Grid.Column>
          <Grid.Column
            width={11}
            verticalAlign="middle"
            style={{ paddingLeft: '0px' }}
          >
            {!!demonWarriorCount && (
              <List>
                {warriors.map((warrior, index) => (
                  <DemonWarriorItem key={index} warrior={warrior} />
                ))}
              </List>
            )}
            {!demonWarriorCount && <span>No Demon warriors in play</span>}
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Segment>
  )
}

GameSummary.propTypes = {
  turn: PropTypes.number.isRequired,
  demonDice: PropTypes.number.isRequired,
  threatDice: PropTypes.number.isRequired,
  warriors: PropTypes.array.isRequired,
}

export default GameSummary
