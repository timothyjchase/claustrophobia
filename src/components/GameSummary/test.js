import React from 'react'
import GameSummary from './index'

describe('<GameSummary />', () => {
  const commonProps = {
    turn: 1,
    demonDice: 2,
    threatDice: 3,
  }
  describe('with a variety of warriors', () => {
    const component = shallow(
      <GameSummary
        {...commonProps}
        warriors={[
          { numberInPlay: 0 },
          { numberInPlay: 1 },
          { numberInPlay: 2 },
        ]}
      />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with no warriors', () => {
    const component = shallow(
      <GameSummary {...commonProps} warriors={[{ numberInPlay: 0 }]} />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
