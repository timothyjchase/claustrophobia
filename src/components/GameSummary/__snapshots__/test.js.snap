// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameSummary /> with a variety of warriors should match snapshot 1`] = `
<Segment
  style={
    Object {
      "padding": "10px 10px 10px 10px",
    }
  }
>
  <Grid
    stackable={false}
  >
    <GridRow
      stretched={true}
    >
      <GridColumn
        verticalAlign="middle"
        width={5}
      >
        <div>
          <strong>
            Turn:
            1
          </strong>
          <br />
          <DieImage
            style={
              Object {
                "display": "inline",
                "height": "30px",
                "margin": "5px 4px 0px 0px",
                "width": "30px",
              }
            }
            type="DEMON"
            value={2}
          />
          <DieImage
            style={
              Object {
                "display": "inline",
                "height": "30px",
                "margin": "5px 0px 0px 4px",
                "width": "30px",
              }
            }
            type="THREAT"
            value={3}
          />
        </div>
      </GridColumn>
      <GridColumn
        style={
          Object {
            "paddingLeft": "0px",
          }
        }
        verticalAlign="middle"
        width={11}
      >
        <List>
          <DemonWarriorItem
            key="0"
            warrior={
              Object {
                "numberInPlay": 0,
              }
            }
          />
          <DemonWarriorItem
            key="1"
            warrior={
              Object {
                "numberInPlay": 1,
              }
            }
          />
          <DemonWarriorItem
            key="2"
            warrior={
              Object {
                "numberInPlay": 2,
              }
            }
          />
        </List>
      </GridColumn>
    </GridRow>
  </Grid>
</Segment>
`;

exports[`<GameSummary /> with no warriors should match snapshot 1`] = `
<Segment
  style={
    Object {
      "padding": "10px 10px 10px 10px",
    }
  }
>
  <Grid
    stackable={false}
  >
    <GridRow
      stretched={true}
    >
      <GridColumn
        verticalAlign="middle"
        width={5}
      >
        <div>
          <strong>
            Turn:
            1
          </strong>
          <br />
          <DieImage
            style={
              Object {
                "display": "inline",
                "height": "30px",
                "margin": "5px 4px 0px 0px",
                "width": "30px",
              }
            }
            type="DEMON"
            value={2}
          />
          <DieImage
            style={
              Object {
                "display": "inline",
                "height": "30px",
                "margin": "5px 0px 0px 4px",
                "width": "30px",
              }
            }
            type="THREAT"
            value={3}
          />
        </div>
      </GridColumn>
      <GridColumn
        style={
          Object {
            "paddingLeft": "0px",
          }
        }
        verticalAlign="middle"
        width={11}
      >
        <span>
          No Demon warriors in play
        </span>
      </GridColumn>
    </GridRow>
  </Grid>
</Segment>
`;
